### DISCUSSION TITLE
Show Slint in VS Code Webview with Input Events and Automatic Code Changes Reflection for Node.js backend without webassembly

### DISCUSSION THREAD

**Discussion by ji-podhead** [slint-ui/slint#4728] [Q&A]
I'm developing a [VS Code extension](https://github.com/ji-soft/fresh_js_slint) that integrates Slint (or a similar tool) into a webview panel in VS Code. The main goal is to enable interactive input events (e.g. mouse clicks and keystrokes) within the web view and is intended to aid in the development of Slint-UIs. This is especially important for a Node.js project where I want to use Nodemon to automatically reflect code changes.

I'm investigating methods to render the output of Slint (or similar tools) directly in the webview, ideally without resorting to WebAssembly, to streamline the development process. My concern with WebAssembly is the longer compilation time and the additional complexity it brings to the project.

I'm looking for advice on the most effective approach to implementation.
What alternative methods to render tools directly in VS Code that may not require WebAssembly.
Do you have any tips for handling input events in a webview, especially in the context of a Node.js project using Nodemon?
Can i publish this like this as my first extension on the vscode marketplace, or should i fork the official slint extension before?
Its my first time contributing to a project like this, so pls keep that in mind :)
i also thought of implementing a method that converts your project between webassembly and nodejs, so you can use the webview for production and then switch back to build with nodejs etc. also maybe adding a method to git pull a slint example/demo project.

this is what it can do so far:
![about](https://github.com/slint-ui/slint/assets/117015142/5d6db8ec-c9b5-493b-b16e-5bd4590835e1)

**Comment by ji-podhead** [comment]
hmm maybe webassembly cant be compiled with nodemon. i think i should look for a different approach and add an additional rust backend in the extension to build the wasm files. or use the official slint extension to show the slint window inside a webview pane and send the input events using it

**Reply by ji-podhead** [reply]
i think ill fork the original extension and implement a method that calls the the callback functions using websockets

**Reply by ji-podhead** [reply]
i think ill add an eventlistener in the wasm_preview.js
like so: 
```// Innerhalb des <script>-Tags in getPreviewHtml
document.addEventListener('DOMContentLoaded', () => {
    document.querySelectorAll('button').forEach(button => {
        button.addEventListener('click', () => {
            const buttonName = button.name || button.textContent;
            vscode.postMessage({ command: 'button_clicked', buttonName });
        });
    });
});
```
and then i call the function via the slint file

**Comment by ogoffart** [comment]
I am not sure I exactly understand what you're trying to do. Could you please elaborate?
(Your link don't work for me, it is perhaps not a public repository)
